internal/apis/kfd/v1alpha2/eks/delete/distribution.go

```diff
@@ -324,82 +363,97 @@ func (d *Distribution) deleteBlockingResources() error {
 	return nil
 }
 
-func (d *Distribution) getLoadBalancers() error {
-	log, err := d.kubeRunner.Get("all", "svc", "-o",
-		"jsonpath='{.items[?(@.spec.type==\"LoadBalancer\")].metadata.name}'")
+func (d *Distribution) getIngressHosts() ([]string, error) {
+	ingrs, err := d.kubeClient.GetIngresses()
 	if err != nil {
-		return fmt.Errorf("error while reading resources from cluster: %w", err)
+		return nil, fmt.Errorf("error getting ingresses: %w", err)
 	}
 
-	reg := regexp.MustCompile(`'(.*?)'`)
+	var hosts []string
 
-	logStringIndex := reg.FindStringIndex(log)
-
-	if len(logStringIndex) == 0 {
-		return fmt.Errorf("%w: error while parsing kubectl get response", errPendingResources)
+	for _, ingress := range ingrs {
+		hosts = append(hosts, ingress.Host...)
 	}
 
-	logString := log[logStringIndex[0]:logStringIndex[1]]
-
-	if logString != "''" {
-		return fmt.Errorf("%w: %s", errPendingResources, logString)
-	}
+	hosts = slices.Uniq(hosts)
 
-	return nil
+	return hosts, nil
 }
 
-func (d *Distribution) getListOfResourcesNs(ns, resName string) error {
-	_, err := d.kubeRunner.Get(ns, resName, "-o",
-		"jsonpath={range .items[*]}{\""+resName+" \"}\"{.metadata.name}\"{\" deleted (dry run)\"}{\"\\n\"}{end}")
-	if err != nil {
-		return fmt.Errorf("error while reading resources from cluster: %w", err)
-	}
-
-	return nil
-}
+func (d *Distribution) getHostedZones() (map[string]string, error) {
+	zones := make(map[string]string)
 
-func (d *Distribution) getIngresses() error {
-	log, err := d.kubeRunner.Get("all", "ingress", "-o", "jsonpath='{.items[*].metadata.name}'")
+	route53, err := d.awsRunner.Route53("list-hosted-zones", "--query", "HostedZones[*].[Id,Name]",
+		"--output", "text")
 	if err != nil {
-		return fmt.Errorf("error while reading resources from cluster: %w", err)
+		return zones, fmt.Errorf("error getting hosted zones: %w", err)
 	}
 
-	reg := regexp.MustCompile(`'(.*?)'`)
+	re := regexp.MustCompile(`/hostedzone/(\S+)\t(\S+)\.`)
 
-	logStringIndex := reg.FindStringIndex(log)
+	matches := re.FindAllStringSubmatch(route53, -1)
 
-	if len(logStringIndex) == 0 {
-		return fmt.Errorf("%w: error while parsing kubectl get response", errPendingResources)
+	for _, match := range matches {
+		zones[match[1]] = match[2]
 	}
 
-	logString := log[logStringIndex[0]:logStringIndex[1]]
+	return zones, nil
+}
 
-	if logString != "''" {
-		return fmt.Errorf("%w: %s", errPendingResources, logString)
+func (d *Distribution) checkPendingDNSRecords(hosts []string, hostedZones map[string]string) error {
```

Comment:
I would rename this to something like an assertion, as it fails if the condition is not verified: it is not simply checking it.
```suggestion
func (d *Distribution) assertEmptyDNSRecords(hosts []string, hostedZones map[string]string) error {
```